extensions/picocli/deployment/src/main/java/io/quarkus/picocli/deployment/PicocliProcessor.java

```diff
@@ -99,6 +104,19 @@ void picocliRunner(ApplicationIndexBuildItem applicationIndex,
             additionalBean.produce(AdditionalBeanBuildItem.unremovableOf(DefaultPicocliCommandLineFactory.class));
             quarkusApplicationClass.produce(new QuarkusApplicationClassBuildItem(PicocliRunner.class));
         }
+
+        // make the arguments of @Command unremovable beans if necessary
+        Collection<AnnotationInstance> commandInstances = index.getAnnotations(CommandLine.Command.class);
+        for (AnnotationInstance commandInstance : commandInstances) {
+            AnnotationValue versionProvider = commandInstance.value("versionProvider");
+            if (versionProvider != null) {
+                ClassInfo versionProviderInfo = index.getClassByName(versionProvider.asClass().name());
+                if (versionProviderInfo != null) {
+                    // TODO: should we make these beans even when they are not annotated as such?
```

Comment:
I don't think so. 

Do we have examples of keeping other things around we don't have any explicit reference to in annotations/config ?

Reply:
I can't remember off hand, but I am pretty we do.

Reply:
the only reason I could see that be relevant is if you use picocli programmatic API and circumvent the annotation setup ...but I don't see that as being the job of Quarkus to make all such possible subclasses unremovable by default for that edge case.